externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.td

```diff
@@ -0,0 +1,65 @@
+#ifndef TORCH_MLIR_DIALECT_TCP_OPS
+#define TORCH_MLIR_DIALECT_TCP_OPS
+
+include "torch-mlir-dialects/Dialect/Tcp/IR/TcpOpsBase.td"
+
+include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+def Tcp_TanhOp : Tcp_UnaryOp<"tanh", [NoSideEffect]> {
+  let summary = "Computes tanh of input, elementwise";
+
+  let description = [{
+    Elementwise unary op to compute tanh.
+  }];
+
+  let arguments = (ins
+    Tcp_Tensor:$in
+  );
+
+  let results = (outs
+    Tcp_Tensor:$out
+  );
+
+  let assemblyFormat = "$in attr-dict `:` type($in) `->` type($out)";
+}
+
+def Tcp_AddOp : Tcp_BinaryOp<"add", [NoSideEffect]> {
+  let summary = "Computes elementwise addition";
+
+  let description = [{
+    Elementwise addition.
+  }];
+
+  let arguments = (ins
+    Tcp_Tensor:$in1,
+    Tcp_Tensor:$in2
+  );
+
+  let results = (outs
+    Tcp_Tensor:$out
+  );
+
+  let assemblyFormat = "$in1 `,` $in2 attr-dict `:` type($in1) `,` type($in2) `->` type($out)";
+}
+
+def Tcp_MatmulOp : Tcp_Op<"matmul", [NoSideEffect]> {
+  let summary = "Computes 2D / 3D matrix multiplication";
+
+  let description = [{
+    Matrix multiplication of two 2D / 3D tensors.
+  }];
+
+  let arguments = (ins
+    Tcp_Tensor2Dor3D:$lhs,
+    Tcp_Tensor2Dor3D:$rhs
```

Comment:
Please add a comment here or in the `description` on why we're restricting to rank 2 and 3 only.

Reply:
Added